pkg/broker/logic.go

```
@@ -321,7 +321,7 @@ func (b *BrokerLogic) deleteResources(planID, instanceID string) error {
 	}
 
 	err = b.DeleteHabitat(name, ns)
-	if err != nil {
+	if err != nil && !k8sErrors.IsNotFound(err) {
```

If the delete operation failed because of a 404, should it note propagate the error anyway? And I'm thinking if how to handle the 404 error should be left to the caller of this method?
For eg: In `Deprovision`, we might just show a friendly message that the instance does not exist. But this won't be possible if we didn't let the error propagate.

What do you think?